src/services/redis.service.ts

```diff
@@ -18,6 +18,39 @@ export class RedisService {
     });
   }
 
+  addTokenToVerificationList(input: AuthInterface.ITokenToBlacklistInput): Promise<number> {
+    return new Promise((res, rej) => {
+      return redisClient.set(input.email, input.token, 'EX', 24*60*60,  function(err, result) {
+        if (err) {
+          rej(err);
+        }
+        res(result);
+      });
+    });
+  }
+
+  get(email: string): Promise<string> {
+    return new Promise((res, rej) => {
+      return redisClient.get(email, function(err, result) {
+        if (err) {
+          rej(err);
+        }
+        res(result);
+      }); 
+    });
+  }
+
+  removeVerificationToken(email: string): Promise<string> {
+    return new Promise((res, rej) => {
+      return redisClient.del(email, function(err, result) {
+        if (err) {
+          rej(err);
+        }
+        res(result);
+      }); 
+    });
+  }
```

Comment:
You could take advantage of async functions that (between other things) takes care of returning a promise for you instead, what do you think?